src/bcf/header.rs

```diff
@@ -180,6 +180,49 @@ impl HeaderView {
 
         Ok((_type, length))
     }
+
+    /// Convert string ID (e.g., for a `FILTER` value) to its numeric identifier.
+    pub fn id2int(&self, id: &[u8]) -> Result<u32, IdError> {
```

Comment:
Please return a newtype `bcf::header::ID` here, see: https://doc.rust-lang.org/stable/rust-by-example/generics/new_types.html. For implementing the newtype, use the crate https://docs.rs/newtype_derive. This allows you to inherit methods from u32.

Reply:
Call this function `name_to_id`.